### PR TITLE
[Method References] LPS-160434 Convert method references to field references in class extends BasePanelApp

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang , I have removed the method setUserNotificationEventLocalService in class NotificationsPanelApp.

(The repush is just to fix the semver issues  in the last pr https://github.com/liferay-core-infra/liferay-portal/pull/1752)

Detial: https://github.com/liferay-core-infra/liferay-portal/pull/1711

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @zhangyan0701  , Could you please help me check the test results ? Thanks!

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java [hunk 1]
```diff
@@ -111,13 +111,6 @@ public List<PanelApp> getPanelApps(String parentPanelCategoryKey) {
...
-				if (portlet == null) {
-					portlet = _portletLocalService.getPortletById(
-						panelApp.getPortletId());
-
-					panelApp.setPortlet(portlet);
-				}
-
 				if (portlet == null) {
 					return false;
 				}
```
Comment (tinatian) [inline modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java:116]: @W2jx @dantewang After we changed the structure of PanelApp, passed in portlet or referenced portlet, which mean once you have an instance of PanelApp, the _portlet attribute should not be null. So I think if we do null checking in the constructor should be enough, and here, we should consider it is never null.

Comment (W2jx) [reply modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java:116]: I will fix it.

**Comment by W2jx** [comment]
Force-push is just to fix the problem mentioned in the comment.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java [hunk 2]
```diff
@@ -381,23 +370,17 @@ public void keyEmitted(
...
-
 			if (panelApp instanceof BasePanelApp) {
 				BasePanelApp basePanelApp = (BasePanelApp)panelApp;
 
 				basePanelApp.setPortletLocalService(_portletLocalService);
 			}
+
+			Portlet portlet = panelApp.getPortlet();
+
+			if (portlet != null) {
```
[the remark this answers is not in the document]
Comment (dantewang) [reply modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java:381]: @W2jx Plz re-trigger the tests after you force push.

**Pull request closed without merging**